keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ClusterHandler.kt

```diff
@@ -42,80 +44,27 @@ class ClusterHandler(
-        "delayBeforeDisableSec" to 0,
-        "delayBeforeScaleDownSec" to 0,
-        "maxRemainingAsgs" to 2,
-        "rollback" to mapOf(
-          "onFailure" to false
-        ),
-        "scaleDown" to false,
-        // </things to do with the strategy>
-        "capacity" to mapOf(
-          "min" to spec.capacity.min,
-          "max" to spec.capacity.max,
-          "desired" to spec.capacity.desired
-        ),
-        "targetHealthyDeployPercentage" to 100, // TODO: any reason to do otherwise?
-        "cooldown" to spec.health.cooldown.seconds,
-        "enabledMetrics" to spec.health.enabledMetrics,
-        "healthCheckType" to spec.health.healthCheckType.name,
-        "healthCheckGracePeriod" to spec.health.warmup.seconds,
-        "instanceMonitoring" to spec.launchConfiguration.instanceMonitoring,
-        "ebsOptimized" to spec.launchConfiguration.ebsOptimized,
-        "iamRole" to spec.launchConfiguration.iamRole,
-        "terminationPolicies" to spec.health.terminationPolicies.map(TerminationPolicy::name),
-        "subnetType" to spec.location.subnet,
-        "availabilityZones" to mapOf(
-          spec.location.region to spec.location.availabilityZones
-        ),
-        "keyPair" to spec.launchConfiguration.keyPair,
-        "suspendedProcesses" to spec.scaling.suspendedProcesses,
-        "securityGroups" to spec.securityGroupIds,
-        "stack" to spec.moniker.stack,
-        "freeFormDetails" to spec.moniker.detail,
-        "tags" to spec.tags,
-        "useAmiBlockDeviceMappings" to false, // TODO: any reason to do otherwise?
-        "copySourceCustomBlockDeviceMappings" to false, // TODO: any reason to do otherwise?
-        "virtualizationType" to "hvm", // TODO: any reason to do otherwise?
-        "moniker" to mapOf(
-          "app" to spec.moniker.application,
-          "stack" to spec.moniker.stack,
-          "detail" to spec.moniker.detail,
-          "cluster" to spec.moniker.cluster
-        ),
-        "amiName" to spec.launchConfiguration.imageId,
-        "reason" to "Diff detected at ${clock.instant().iso()}",
-        "instanceType" to spec.launchConfiguration.instanceType,
-        "type" to "createServerGroup",
-        "cloudProvider" to CLOUD_PROVIDER,
-        "loadBalancers" to spec.dependencies.loadBalancerNames,
-        "targetGroups" to spec.dependencies.targetGroups,
-        "account" to spec.location.accountName
-      )
-
-      cloudDriverService.getAncestorServerGroupName(spec)
-        ?.let { ancestorServerGroup ->
-          job["source"] = mapOf(
-            "account" to spec.location.accountName,
-            "region" to spec.location.region,
-            "asgName" to ancestorServerGroup
-          )
-          job["copySourceCustomBlockDeviceMappings"] = true
-        }
+      val spec = resource.spec
+      val job = when {
+        diff.isCapacityOnly() -> spec.resizeServerGroupJob()
+        else -> spec.createServerGroupJob()
+      }
```

Comment:
Determines here which type of job to submit to Orca

Reply:
That is cool!

Reply:
Right?